packages/usa-date-picker/src/index.js

```diff
@@ -1801,8 +1809,18 @@ const handleMouseoverFromDate = (dateEl) => {
   if (hoverDate === currentCalendarDate) return;
 
   const dateToDisplay = parseDateString(hoverDate);
-  const newCalendar = renderCalendar(calendarEl, dateToDisplay);
-  newCalendar.querySelector(CALENDAR_DATE_FOCUSED).focus();
+  const focusedEl = document.activeElement;
+
+  // If the calendar contains a focused element, reapply focus after new calendar renders.
+  // Targeted using aria-label because they are unique and prevents error when
+  // focus is in the table header.
+  if (calendarEl.contains(focusedEl)) {
+    const focusElLabel = focusedEl.getAttribute("aria-label");
+    const newCalendar = renderCalendar(calendarEl, dateToDisplay);
+    newCalendar.querySelector(`[aria-label="${focusElLabel}"]`).focus();
```

Comment:
## aria-label vs. dataset.value

In order to refocus the correct button, I used aria-label because they were unique and included on all buttons on the calendar. 

This also prevented an error when tabbing to navigation, month, and year buttons which did not include a `data-value` attribute